src/dbnode/storage/series/buffer.go

```diff
@@ -682,27 +682,27 @@ func (b *dbBuffer) FetchBlocksForColdFlush(
 	start time.Time,
 	version int,
 	nsCtx namespace.Context,
-) ([]xio.BlockReader, error) {
+) (block.FetchBlockResult, error) {
 	res := b.fetchBlocks(ctx, []time.Time{start},
-		streamsOptions{filterWriteType: true, writeType: ColdWrite, nsCtx: nsCtx})
+		streamsOptions{filterWriteType: true, writeType: ColdWrite, nsCtx: nsCtx}, true)
```

Comment:
Can we put the third param here into `streamOptions` - I think it belongs with the other orgs here, in that it affects what is returned on a per stream/block result that comes back.